server.hpp

```diff
@@ -154,6 +158,7 @@ class Room {
             break;
         }
         case OpCode::GIVEUP_OP: {
+            if(data2!="b"&&data2!="w") logger->debug("Receive GIVEUP_OP with invalid role {}",data2);
```

Comment:
Maybe we can trace more details like printing player info (role, name, etc). `to_string` method of `Player` is needed.
And this line can be deleted since we have printed `role`.
Also, trace the message before any processing as well.